src/AppInstallerCLICore/Workflows/GetRequestedUrlFromManifest.cpp

```diff
@@ -0,0 +1,71 @@
+#include "pch.h"
+#include "GetRequestedUrlFromManifest.h"
+
+#include "ManifestComparator.h"
+#include "ShellExecuteInstallerHandler.h"
+
+using namespace AppInstaller::CLI;
+using namespace AppInstaller::Utility;
+using namespace AppInstaller::Manifest;
+
+namespace AppInstaller::CLI::Workflow
+{
+	void GetRequestedUrlFromManifest(Execution::Context& context)
+	{
+		if (context.Contains(Execution::Data::SearchResult)) {
+			const auto& searchResult = context.Get<Execution::Data::SearchResult>();
+			if (searchResult.Matches.size() > 1)
+			{
+				context.Reporter.Warn() << "More than one result was found please refine your query. Use --Id, --Name --Tag" << std::endl;
+			}
+			else
+			{
+				if (context.Contains(Execution::Data::Manifest)) {
+					const auto& manifest = context.Get<Execution::Data::Manifest>();
+					Workflow::ManifestComparator manifestComparator(context.Args);
+					const auto selectedLocalization = manifestComparator.GetPreferredLocalization(manifest);
+					if (context.Args.Contains(Execution::Args::Type::Homepage)) {
+						if (selectedLocalization.Homepage.empty())
+						{
+							context.Reporter.Info() << "No Homepage url found within manifest" << std::endl;
```

Comment:
>"No Homepage url found within manifest" [](start = 34, length = 39)

Any string that is output to the Reporter should be a resource to allow for localization.  I haven't gotten quite far enough down the path to make that a compilation error, but I don't want to add any debt :)